aten/src/ATen/native/mobile/cpu/internal/Add.cpp

```diff
@@ -0,0 +1,244 @@
+#include <ATen/native/mobile/cpu/Engine.h>
+
+#ifdef USE_XNNPACK
+
+#include <ATen/native/mobile/cpu/internal/Allocator.h>
+#include <ATen/native/mobile/cpu/internal/Common.h>
+
+namespace at {
+namespace native {
+namespace mobile {
+namespace cpu {
+namespace internal {
+namespace add {
+
+struct Context final {
+  Operator add_op;
+
+  static constexpr float kMin = -std::numeric_limits<float>::infinity();
+  static constexpr float kMax = std::numeric_limits<float>::infinity();
+};
+
+namespace {
+
+// Supports NHWC and NCHW FP32 adds.
+
+bool available(
+    const int64_t input1_channels,
+    const int64_t input2_channels,
+    const int64_t output_channels,
+    const Scalar output_min,
+    const Scalar output_max) {
+         // Mobile
+  return mobile::cpu::available() &&
+         // Input1
+         (input1_channels > 0) &&
+         // Input2
+         (input2_channels > 0) &&
```

Comment:
The `> 0` tests on channels seem a little pointless, since they all be handled at the very top if we have logic to ensure that zero numel tensors are handled correctly (though I am a little surprised that XNNPACK doesn't handle this correctly)

Reply:
I'll remove them.